### ISSUE TITLE
How to delete an inverter with all its sensors from the grott integration page ?

### ISSUE THREAD

**Issue by typxxi** [muppet3000/homeassistant-grott#83]
I might have made a mistake in a multi inverter environmen cause I had not setup the grott.ini properly or better I had not rebooted the grott system BEFORE adding a SPH inverter to my HA by pointing the ip adress in the sph stick to my grott server.

Then the grott ha integration identified a new inverter but i guess it got the mapping wrong cause I can see from the cloud web ui clearly the production of today.

So this figure and the total should be found in the right sensor I bet but i found them rather in the dc output of a string, so i assume I caused a kind of mismapping due to the grott.ini which was not properly maintained or the missing reboot.

THerefore I wanted to delete this particular device but for what ever reason I can not find a DELETE DEVICE option, not via 3 dots menu nor right click.

And no: deleting the integration and setting it up again is not alternativ due to 8 inverters I would have to setup properly without any typo to match the sensors name used in charts and dashboards. 

So I might have made a mistake and overlooked the delete button or the delete button might be missing.

Hopefully someone knows a way how to delete such device - or would I simply go into the developer pane and search for all the sensors that belong to "hybrid" cause that is part of the name. I will try that way but maybe worth to think about a "DELETE BUTTON" cause 34 sensors are a lot and if things go wrong in a mulit inverter environment there are hundreds of sensors involved - at least here 280 or so.

**Comment by typxxi** [comment]
I have tried to open and delete the sensors belonging to the hybrid but I can not cause there is no such option.
Then I have deactivated the inverter in the grott ha configuration , but then the sensors completely disappeared in the developer tools pane. 
I then opened the device from the integration where I can see all sensors again and tried to delete on.
That did not work out at all.

Now I will reboot my grott server and hope that this might solve my mapping issue.

**Comment by muppet3000** [comment]
Hi @typxxi I'm not sure I ever considered the 'delete' option. In fact I'm not even sure if Home Assistant's integration interface even supports it. I can have a look though.
One option I can think of is to use the 'reconfigure' option for the integration to list only the serial numbers of the inverters that you DO want to see and exclude the SPH system you're trying to avoid getting metrics for.

**Comment by typxxi** [comment]
Hi Chris,

thanks for the reply which arrived a minute after I had solved the issue cause last night I had tried to dive deeper and checked all the directories to find an access point.

ATTENTION: I have had a different issue. I did not wanna exclude the SPH inverter!
I had made a mistake many will make that later expand their solar power system with new panels and therefore will add a new inverter which requires a few steps mentioned below you might add to your documentation on the top level cause otherwise people will tend to strand here where I had stranded. 

My mistake was to not maintain the grott.ini file properly and add each and every inverter in the mapping line. 
From there grott ha integration got a wrong mapping cause I could find the data from the app / web ui in different and most likely wrong sensors. but as always: if I throw garbage in I might get even get more garbage out.

THEREFORE MY THOUGHT: DELETE THE DEVICE / INVERTER which I could not and the only way around without a **"delete this device" option** was a complete delete of the grott integration and not a ban of my sph inverter in general (which would be easy by repointing the stick to growatt server instead of my pi 3 grott proxy).


**I. "my solution or workaround"**#

DELETE GROTT CONFIGURATION AND CONFIGURE FROM SCRATCH is the short version but here in a step by step way for all those who might be struggling there again and want avoid making mistakes, so a "a proper way" which meant:

#1 I went to the integration page
#2 I **opened GROTT integration** and 
#3 there I opened **each device and copied its name from there to a notepad file** (cause I have a lot of devices which names have to match after that)
#4 Now I choose from within the grott integration via the **3 dots menu DELETE**, which **deleted ALL inverters and ALL SENSORS** 
#5 after that I had thought GROTT HA integration would be completely gone but I still ended up in the grott integration which now was empty 
#6 then I finally pressed **CONFIGURE** and you know the magic your grott integration is doing ?

Right, all 8 devices were back within 2 minutes but of cause now with their serials and not with the names I had given them which I had saved in step 2.


#7 Now I had to rename all these 8 inverters based on my notepad file and they then matched all my dashboard references 

**FOR ALL OTHERS FOLLOWING THIS WAY MAKE SURE THAT IN CASE YOU RENAME THE DEVICE that you do the proper decision when asked if you also wanna rename the sensors according to the device name**.

I always do that which means my sensor names start with the device name so make sure to have a good naming sheme that you will not have to change. Easy ones use just an inverter related one, but in case of multiple inverter with 2 or more strings you might consider an inverter name which also speaks for its strings. Makes it a lot easier to have the overview at least in the beginning and in case you are optimizing strings and inverters in the beginning. Then again: you can rename the inverter string thing like ours S1: W1.1 + W1.2 

S1 means surplus inverter 
W1.1 stands for: 
#1 west half of the southern roof 
#2 first row from the top and then 
#3 for 1st string and 2nd string where first is on the side where the sun goes up, so east.

You might wonder what surplus means?
You have a solar power system on your roof or fence or in the garden that produces energy primarily for use in your home and you feed the surplus energy into the grid for about 8 cents.
The opposite is a dedicated solar power system which is called a full feeder and means you are not using the kWh produced but selling it to the grid operator and get 13 cents instead)
Therefore you can have here 2 kind of solar power systems on your roof and you can choose every year how to use each of these systems as a whole, so you can turn the currently full feeder into the surplus feeder billing when you build a bigger solar power array and you will get later a heatpump and car cause in case we sell for 13 cent or 8 cent we need to buy back during night or so for about 30 cent. In case you know your consumption a small full inverter can make sense money wise.  

**II. "your solution"**#
I was not aware that the reconfigure would achieve the same result and I am also still not sure if it really can cause: 

**Does YOUR RECONFIGURE actually delete the old sensors first** before starting to add the new ones ?
I doubt that cause I have used the function before here
![image](https://github.com/muppet3000/homeassistant-grott/assets/20662430/20ea6649-4caa-40c2-8476-50952fa71c5c)

which literally added new inverters (before unknown) to the integration, but I remember for sure that it did not delete any of the previously installed cause they were still there and still with the name i had given to them.   

**III. a possible conclusion**#

FINALLY: 
`list only the serial numbers of the inverters that you DO want to see and exclude the SPH system you're trying to avoid getting metrics for.` 

No, I did not want that and I guess that it would not work cause you must have a kind of autodetection cause my sph4600 had been integrated without being in that mapping line so grott did not care about that line and your integration also did not care.
And I guess it could / should work this way cause in case you do not wanna integrate 1 inverter the proper way is to let its stick report to growatt and not to redirect it to the grott server. 

My troubles began when I have had added my last inverter which was the first and only SPH Hybrid here.
I had not updated my GROTT.ini regarding the inverter and serial mapping line. 

As usually I went the same way which had worked before for all different MiC and MiN inverters and hoped for the best. And Johan had implemented a year or 2 ago and automatic detection how to map an inverter by its protocol or structure which had save me before cause I had not maintained the grott.ini line for multi inverter mapping properly. But grott had a good algorithm and therefore I did not face any issues cause Johans default algorithm had worked out great, but SPH as a Hybrid is a different beast and therefore went terribly wrong which I recogniced yesterday or so. 

**THEREFORE:** I guess you should add a line to your introduction and documentation that in case of multi inverter environment you have to maintain the grott.ini properly regarding the mapping BEFORE you add another inverter which means before changing the inverter's wifi stick IP destination to the grott server. 

grott.ini requires for a good mapping of the data to the HA sensors this line in the generic section (attention: johan is still refering that it belongs to a [system] section which he must have removed or renamed cause somewhere on his site I had found a reference that the multi inverter line belongs to there. I emphasize that cause when I had appended that multi inverter line  to the [mqtt] section it had caused an "target refused data" error in the grott terminal which I could see appear constantly last night when the hybrid was still reporting. 

I guess many people will tend to forget to update and maintain the grott.ini first and before changing the ip of the stick- as I did. And therefore they will only remember to reconfigure the inverter stick target IP adress and then cause issues with new inverters (beyond the mic and min inverter which seem to work out of the box as Johan had mentioned somewhere). 

But for general awareness reasons I would suggest to add a more general line for every user to emphasize that in case grott got an issue with autodetection which inverter is used it is better to maintain the inverter serial list even for 1 single inverter.

and
"if you later wanna add an additional inverter always make sure to add to the [mqtt] section 
`mqttinverterintopic="True"`
and to add to the [generic section]
`invtypemap = {"inverter serial 1": "MIN", "inverter serial 2": "MIN", "inverter serial 3": "SPH"}`

I suggest to emphasize for a good behaviour to do that even for the first one cause then people will less likely tend to forget and not fell into the trap of "default config will work". If I had done for the first one I guess I would have remembered it easier, but again, it had worked for the first 7 all min and mic so why should it not work for the 8th ?
And here I am with number 8 not working properly or as expected. 

And if you do not emphasize on maintaining the mapping line in the grott.ini every diy growatt installation might end up in the same mess I had faced once they decide to add a few more panels and another inverter. So I guess it might be worth to emphasize that your grott ha integration needs a proper maintained grott.ini and the inverter serial mapping line is key for that - even for 1 inverter.

here my mismapping screenshot with phantasy values if you look at the big figures of a brand new device with just 9 kWh production, but some crazy numbers in it, but also AC voltage missing. 
![image](https://github.com/muppet3000/homeassistant-grott/assets/20662430/ea84c3be-1ddf-4e53-bdaf-24bdfa0d448c)

**IV. "Is grott ha integration SPH 4600 missing sensors ?"**#
 
Might be OT and belongs to a new issue but I just wanna ask where I have to open such topic, grott or grott ha integration ?

The sph 4600 seems to have more data / sensors (according to grott terminal) **than grott ha is showing** 
I hope this is a full data set from grott about a message received and transmitted (to growatt and HA).

```
	 - Growatt packet received:
		  <socket.socket fd=10, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('192.168.178.28', 56606), raddr=('47.91.67.66', 5279)>
	 - Growatt original Data:
		 \x04\x76\x00\x06\x03\x3f\x01\x04\x0d\x22\x2c\x4f\x22\x3f\x43\x77\x3e\x23\x77
		 \x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x21\x27\x22\x27\x36\x0c\x22\x5f\x43\x50\x74\x74\x47\x72\x6f\x77\x61\x74\x74
		 \x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x79\x77\x67\x44\x62\x56\x74\x61
		 \x74\x74\x3b\x72\x6a\x77\x61\x43\x0c\x4e\xb8\x6f\x4f\x61\x74\x43\x39\x72\x6f
		 \x77\x61\x74\x74\x47\x72\x6f\x76\x21\x54\x7d\xe7\x72\x6f\x7f\xe4\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x41
		 \x3e\x67\xf8\x4e\x45\x6f\x4a\x61\x74\x42\xcd\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x6d\x74
		 \x47\x72\x15\x76\x69\x8a\xef\x47\x72\x6f\x6f\x61\x74\x74\x32\x72\x6f\x77\x61
		 \x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f
		 \x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x14\x75
		 \x28\x46\x40\x6e\x13\x61\x74\x74\x47\x62\xd6\x67\xb8\x74\x74\x47\x72\x6f\x77
		 \x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x6f\x77\x60\x74\x74\x47\x70\x6f\x77\x61\x74\x74\x47\x72\x6e\x77\x61\x74\x74
		 \x47\x72\x6f\x77\x61\x74\x77\xaf\x76\x0b\x77\x64\x74\x74\x47\x72\x6f\x77\x63
		 \x70\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f
		 \x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x59\xd0
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x57\xd3\x61\x74\x74\x47\x72\x6e\x70
		 \x67\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x6d
		 \x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74
		 \x5f\x72\x6f\x77\x37\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61
		 \x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f
		 \x77\x62\x9c\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77
		 \x61\x74\x74\x46\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x76\x0a\x73\x80\x74\x75
		 \x47\x72\x6f\x75\x61\x74\x74\x47\x76\x4e\x77\x61\x74\x74\x47\x72\x6f\x77\x61
		 \x74\x74\x47\x72\x6f\x77\x78\x74\x74\x47\x07\x6f\x77\x61\x6d\x74\x47\x72\x1a
		 \x77\x61\x4e\x88\x47\x72\x55\x8b\x61\x74\x74\x5f\x72\x6f\x77\x61\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77
		 \x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74
		 \x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x46\x72\x6f\x77\x61
		 \x74\x75\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f
		 \x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47
		 \x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77
		 \x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x6f\xfe\xcc
	 - Grott automatic protocol detection
	 - Grott data record length 839
	 - layout   :  T060104X
	 - no matching record layout found, try generic
	 - Record layout used :  T06NNNNX
	 - Growatt data decrypted V2
	 - Grott Growatt data decrypted
	 - Growatt plain data:
		 04760006033f01044a504338434b37304c4c00000000000000000000000000000000000000004
		 e504353424b503034310000000000000000000000000000000000000000180313031039030000
		 007c00050000377809ca00380000377e00000000000000000001402009a000000885000000000
		 00000000000000000000000000000000000000000000000000000000000000000000000365f13
		 8c0937003d0000368a00000000000000000000000000000000000000000000000000190000007
		 a0108fe9b00000018000000750000000000000000000000000000000000000000000000000000
		 00000000000000000000000000000000000000000000000000000000000000000075015c01320
		 1640000000010b910d90000000000000000000000000000000000000000000000000000000100
		 00000200000000000000010000000000000000000003e80464000500000000000002040000000
		 00000000000000000000000000000000000000000000000000000000000000000000000000000
		 00000000000000000000000038a40000000000000000000038a40000000000010706000000000
		 00000000000000000000000001f00000000000000000000000000000000000000180000005600
		 00000000000000000000000000000000000000000000000000000000000000000003e80000000
		 00000000000000000000000000000000000000000000000000000000000000000000000000000
		 00000000000000000000000000000001000000000000000000000000000000000000000000000
		 00000000000046504e10001000000020000000004210000000000000000000000000000001900
		 000075000000190000007500003afc00003afc000000180000000000000000000000000000000
		 00000000000000000000000000000000000000000000000000000000000000000000000000000
		 00000000000000000000000000000000000000000000000000000000000000000000000000000
		 00000000001000000000001000000000000000000000000000000000000000000000000000000
		 00000000000000000000000000000000000000000000000000000000000000000000000000000
		 00000000000000000000000000000000000000000000000000000000089ad
	 - Possible Inverter serial NPCSBKP041
	 - Matched inverter serial to inverter type SPH
	 - Growatt new layout processing
		 - decrypt       :  True
		 - offset        :  6
		 - record layout :  T06NNNNXSPH

	 - Grott data record date/time processing started
	 - date-time:  2024-03-19T03:16:57
	 - Grott values retrieved:
		 -  datalogserial        :  JPC8CK70LL
		 -  pvserial             :  NPCSBKP041
		 -  pvstatus             :  5
		 -  pvpowerin            :  1420.0
		 -  pv1voltage           :  250.6
		 -  pv1current           :  5.6
		 -  pv1watt              :  1420.6
		 -  pv2voltage           :  0.0
		 -  pv2current           :  0.0
		 -  pv2watt              :  0.0
		 -  pvpowerout           :  1391.9
		 -  pvfrequentie         :  50.0
		 -  pvgridvoltage        :  235.9
		 -  pvgridcurrent        :  6.1
		 -  pvgridpower          :  1396.2
		 -  pvgridvoltage2       :  0.0
		 -  pvgridcurrent2       :  0.0
		 -  pvgridpower2         :  0.0
		 -  pvgridvoltage3       :  0.0
		 -  pvgridcurrent3       :  0.0
		 -  pvgridpower3         :  0.0
		 -  totworktime          :  2412.0
		 -  eactoday             :  2.5
		 -  pvenergytoday        :  2.5
		 -  eactotal             :  12.2
		 -  epvtotal             :  11.7
		 -  epv1today            :  2.4
		 -  epv1total            :  11.7
		 -  epv2today            :  0.0
		 -  epv2total            :  0.0
		 -  pvtemperature        :  34.8
		 -  pvipmtemperature     :  30.6
		 -  pvboosttemp          :  35.6
		 -  bat_dsp              :  0.0
		 -  eacharge_today       :  0.1
		 -  eacharge_total       :  0.2
		 -  batterytype          :  1
		 -  uwsysworkmode        :  5
		 -  systemfaultword0     :  0
		 -  systemfaultword1     :  0
		 -  systemfaultword2     :  0
		 -  systemfaultword3     :  516
		 -  systemfaultword4     :  0
		 -  systemfaultword5     :  0
		 -  systemfaultword6     :  0
		 -  systemfaultword7     :  0
		 -  pdischarge1          :  0.0
		 -  p1charge1            :  0.0
		 -  vbat                 :  0.0
		 -  SOC                  :  0
		 -  pactouserr           :  0.0
		 -  pactousertot         :  0.0
		 -  pactogridr           :  0.0
		 -  pactogridtot         :  0.0
		 -  plocaloadr           :  1450.0
		 -  plocaloadtot         :  1450.0
		 -  spdspstatus          :  0.1
		 -  spbusvolt            :  179.8
		 -  etouser_tod          :  0.0
		 -  etouser_tot          :  0.0
		 -  etogrid_tod          :  0.0
		 -  etogrid_tot          :  3.1
		 -  edischarge1_tod      :  0.0
		 -  edischarge1_tot      :  0.0
		 -  eharge1_tod          :  0.0
		 -  eharge1_tot          :  0.0
		 -  elocalload_tod       :  2.4
		 -  elocalload_tot       :  8.6
	 - MQTT jsonmsg: 
			 {"device": "NPCSBKP041", "time": "2024-03-19T03:16:57", "buffered": "no",
			 "values": {"datalogserial": "JPC8CK70LL", "pvserial": "NPCSBKP041",
			 "pvstatus": 5, "pvpowerin": 14200, "pv1voltage": 2506, "pv1current": 56,
			 "pv1watt": 14206, "pv2voltage": 0, "pv2current": 0, "pv2watt": 0,
			 "pvpowerout": 13919, "pvfrequentie": 5004, "pvgridvoltage": 2359,
			 "pvgridcurrent": 61, "pvgridpower": 13962, "pvgridvoltage2": 0,
			 "pvgridcurrent2": 0, "pvgridpower2": 0, "pvgridvoltage3": 0,
			 "pvgridcurrent3": 0, "pvgridpower3": 0, "totworktime": 17366683, "eactoday":
			 25, "pvenergytoday": 25, "eactotal": 122, "epvtotal": 117, "epv1today": 24,
			 "epv1total": 117, "epv2today": 0, "epv2total": 0, "pvtemperature": 348,
			 "pvipmtemperature": 306, "pvboosttemp": 356, "bat_dsp": 0, "eacharge_today":
			 1, "eacharge_total": 2, "batterytype": 1, "uwsysworkmode": 5,
			 "systemfaultword0": 0, "systemfaultword1": 0, "systemfaultword2": 0,
			 "systemfaultword3": 516, "systemfaultword4": 0, "systemfaultword5": 0,
			 "systemfaultword6": 0, "systemfaultword7": 0, "pdischarge1": 0, "p1charge1":
			 0, "vbat": 0, "SOC": 0, "pactouserr": 0, "pactousertot": 0, "pactogridr": 0,
			 "pactogridtot": 0, "plocaloadr": 14500, "plocaloadtot": 14500,
			 "spdspstatus": 1, "spbusvolt": 1798, "etouser_tod": 0, "etouser_tot": 0,
			 "etogrid_tod": 0, "etogrid_tot": 31, "edischarge1_tod": 0,
			 "edischarge1_tot": 0, "eharge1_tod": 0, "eharge1_tot": 0, "elocalload_tod":
			 24, "elocalload_tot": 86}}
	 - Grott MQTT topic used : energy/growatt
	 - MQTT message message sent
```

json looks like a kind of summary which then would mean it is a grott topic, right ?
Cause your ha integration creates for every json / mqtt message topic its own sensor I assume.

Sorry, but this is how far I had understood the bits and pieces I had read tonight before I deleted my Grott HA integration configuration and rebuild it from scratch.

**Comment by muppet3000** [comment]
Thanks for the comprehensive breakdown here @typxxi I'll be honest, there's a LOT to process and I haven't been able to go through it all.
It looks like you got to the bottom of your issue and it's now working as you'd hoped.
If there's some specific items you think should be added to the setup steps, please let me know and I'll add them in.
Also, if you're still having issues with the SPH type and want to log a new bug for it we can deal with that separately.

**Comment by typxxi** [comment]
Hi Chris,
yes, it is a long summary but at least this way I can remember what I had done why and how it went wrong. 

Being neglectant once when it worked like the min and mic inverter in the beginning made me more neglectant when I added SPH.

So far grott ha feels rock solid and in case I need something I guess I will ask or write again.

today I had another trouble with grott you might not have but is good to now: 

If you run grott on a pi and grott has been manually installed and you start with grott you will NOT ALWAYS call or start grott from within the usual /home/pi/grott directory

In case you do so guess what mistake you will get and would never believe the cause ?

Well, in case you start grott from /home or /home/pi/desktop you will no longer get any data inside ha 
Grott will receive the inverter messages , then forward these to growatt and also try to send them via mqtt but  you will get

```
	 - Grott MQTT topic used : energy/growatt
	 - MQTT connection refused by target
	 - Grott Send data to PVOutput disabled 
	 - Grott Send data to Influx disabled 
	 - Grott extension processing disabled 
```

**MQTT connection refused by target** is caused simply by starting the script not from within the grott directory.
Took me hours to figure it out cause I had seen it before some time and then it had reappeared everything worked great again.

And then this morning I got the constant message refused by target, where even changing and removing mqtt password did not work out ... and at some point it was running again and I remembered that I had only changed into the grott dir. 

to be clear : being in /home and then typing
python3 /home/pi/grott/grott.py -v
will start grott but cause the refuse error.

Of cause only hits those that run grott manually like I do cause I am still in test phase but I could not believe that. 
Therefore grott ha integration feels rock solid once you got into it.

You might have a look if you could put "delete inverter / device with all its sensors" on your long time goal list cause it will make things easier and especially for all those who have a multi inverter environment cause there you have a lot of work and preparation to do to get it fixed. Just to let you know that I will close this issue cause I can live with my workaround and you can later decide if you find an easy way to implement it. 

Thanks for giving this issue your attention !

**Comment by typxxi** [comment]
Workaround described above for all those struggling at this point !
